agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -234,245 +134,318 @@ public CallReplacementTransformer() {}
     try {
       // Get the class information
       ClassGen cg = new ClassGen(c);
-      ifact = new InstructionFactory(cg);
-
-      map_calls(cg, className, loader, current_method_map);
-
-      JavaClass njc = cg.getJavaClass();
-      if (debug) njc.dump("/tmp/ret/" + njc.getClassName() + ".class");
-
-      if (true) {
+      if (mapCalls(cg)) {
+        if (MapCallsAgent.debug) {
+          JavaClass njc = cg.getJavaClass();
+          njc.dump("/tmp/ret/" + njc.getClassName() + ".class");
+        }
         return (cg.getJavaClass().getBytes());
       } else {
-        debug_transform.log("not including class %s (filtered out)", className);
+        debug_transform.log("class %s not transformed", className);
         return null;
       }
-
     } catch (Throwable e) {
-      out.format("Unexpected error %s in transform", e);
+      System.out.format("Unexpected error %s in transform", e);
       e.printStackTrace();
-      return (null);
+      return null;
     }
   }
 
   /**
-   * Processes each method in cg replacing any specified calls with static user calls.
+   * Indicates whether the named class is in either the AWT ({@code java.awt}) or Swing ({@code
+   * javax.swing}) packages.
    *
-   * @param fullClassName must be packageName.className
+   * @param classname the fully-qualified class name, must be non-null
+   * @return true if the method is in either the AWT or Swing package, false otherwise
    */
-  private boolean map_calls(
-      ClassGen cg,
-      String fullClassName,
-      ClassLoader loader,
-      Map<MethodDef, ReplacementClass> current_method_map) {
+  private boolean isAWTSwingClass(String classname) {
+    return classname.startsWith("java.awt") || classname.startsWith("javax.swing");
+  }
 
-    boolean transformed = false;
+  /**
+   * Indicate whether the class is boot loaded. Checks if the loader or the parent of the loader is
+   * {@code null}.
+   *
+   * @param loader the class loader for the method
+   * @param fullClassName the fully-qualified class name of the method
+   * @return true if the class is boot loaded, false, otherwise
+   */
+  private boolean isBootClass(ClassLoader loader, String fullClassName) {
+    if (loader == null) {
+      return true;
+    } else if (loader.getParent() == null) {
+      return true;
+    }
+    return false;
+  }
 
-    try {
-      pgen = cg.getConstantPool();
+  /**
+   * Indicates whether the named class occurs in a package that is excluded. Tests whether one of
+   * the excluded package names is a prefix of the fully-qualified class name.
+   *
+   * @param fullClassName the fully-qualified class name, must be non-null
+   * @return true if any excluded package is a prefix of the class name, false otherwise
+   */
+  private boolean isExcludedClass(String fullClassName) {
+    for (String excludedPackage : excludedPackages) {
+      if (fullClassName.startsWith(excludedPackage)) {
```

Comment:
I'm not sure this is in the specification.

I changed `MapCallsAgent.loadExclusions()` so that it looks for a period at the end of the string and if there is none, it adds one.